tests/qml/tst_VPNTextArea.qml

```diff
@@ -0,0 +1,83 @@
+import QtQuick 2.3
+import QtTest 1.0
+
+import components.forms 0.1
+
+Item {
+    width: 600;
+    height: 800;
+
+    VPNTextArea {
+        id: vpnTextAreaTest
+    }
+
+    TestCase {
+        name: "VPNTextArea"
+        when: windowShown
+
+        function resetTextArea() {
+            vpnTextAreaTest.textInput.text = "";
+            vpnTextAreaTest.textInput.forceActiveFocus();
+        }
+
+        function insertSingleChar(character) {
+            keyClick(character);
+        }
+
+        function insertMultipleChars(chars) {
+            [...chars].forEach(character => {
+                insertSingleChar(character);
+            });
+        }
+
+        function fillTextInput() {
+            vpnTextAreaTest.textInput.text = "a".repeat(vpnTextAreaTest.textInput.maxCharacterCount);
```

Comment:
It would be nice to have tests to support multi-bytes unicode chars.  Interesting reading: https://hsivonen.fi/string-length/